src/sparkle/reader/table_reader.py

```diff
@@ -0,0 +1,96 @@
+    configurations and provides utility methods to access the fully qualified table name.
+
+    Attributes:
+        spark (SparkSession): The Spark session used for reading data.
+        database_name (str): The name of the database containing the table.
+        table_name (str): The name of the table to read.
+        catalog_name (str): The name of the catalog containing the table. Defaults to "glue_catalog".
+        catalog_id (Optional[str]): The catalog ID, if applicable. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        spark: SparkSession,
+        database_name: str,
+        table_name: str,
+        catalog_name: str = "glue_catalog",
+        catalog_id: str | None = None,
+    ):
+        """Initializes a TableReader instance.
+
+        Args:
+            spark (SparkSession): The Spark session used for reading data.
+            database_name (str): The name of the database containing the table.
+            table_name (str): The name of the table to read.
+            catalog_name (str, optional): The name of the catalog containing the table.
+                Defaults to "glue_catalog".
+            catalog_id (Optional[str], optional): The catalog ID, if applicable. Defaults to None.
+        """
+        self.spark = spark
+        self.database_name = database_name
+        self.table_name = table_name
+        self.catalog_name = catalog_name
+        self.catalog_id = catalog_id
+
+    @classmethod
+    def with_config(
+        cls, spark: SparkSession, config: Config, **kwargs
+    ) -> "TableReader":
+        """Creates a TableReader instance using a configuration object.
+
+        Args:
+            spark (SparkSession): The Spark session used for reading data.
+            config (Config): The configuration object containing table input configuration.
+            **kwargs: Additional keyword arguments passed to the TableReader initializer.
+
+        Returns:
+            TableReader: An instance of TableReader configured with the provided settings.
+
+        Raises:
+            ValueError: If the input configuration is missing in the provided config.
+        """
+        if not config.hive_table_input:
+            raise ValueError("Hive input configuration is missing.")
+        return cls(
+            spark=spark,
+            database_name=config.hive_table_input.database,
+            table_name=config.hive_table_input.table,
+            catalog_name=config.hive_table_input.catalog_name,
+            catalog_id=config.hive_table_input.catalog_id,
+            **kwargs,
+        )
+
+    @property
+    def qualified_table_name(self) -> str:
+        """Gets the fully qualified table name.
+
+        Returns:
+            str: The fully qualified table name in the format "catalog_name.database_name.table_name".
+        """
+        return f"{self.catalog_name}.{self.database_name}.{self.table_name}"
```

Comment:
Should we consider this format only works/is required for Iceberg tables?

Reply:
Look at test `test_read_table`. A normal table also can use this format.

Reply:
Have look at `test_read_table` test. Non-Iceberg table is also working with this format.